### PR TITLE
[K8S] Support to cleanup the spark driver pod after application terminates for retain period

### PR DESCRIPTION
**Pull request by turboFei**
# :mag: Description

## Describe Your Solution 🔧

As title, support to cleanup the application pod after application terminates for retain period 

## Types of changes :bookmark:

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Test Plan 🧪

Test locally.

---

# Checklists
## 📝 Author Self Checklist

- [x] My code follows the [style guidelines](https://kyuubi.readthedocs.io/en/master/contributing/code/style.html) of this project
- [x] I have performed a self-review
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] This patch was not authored or co-authored using [Generative Tooling](https://www.apache.org/legal/generative-tooling.html)

## 📝 Committer Pre-Merge Checklist

- [x] Pull request title is okay.
- [x] No license issues.
- [x] Milestone correctly set?
- [ ] Test coverage is ok
- [x] Assignees are selected.
- [x] Minimum number of approvals
- [ ] No changes are requested

**Be nice. Be informative.**

### CODE REVIEW

File: kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala [hunk 1]
```diff
@@ -1231,6 +1231,14 @@ object KyuubiConf {
       .checkValue(_ > 0, "must be positive number")
       .createWithDefault(Duration.ofMinutes(5).toMillis)
 
+  val KUBERNETES_TERMINATED_APPLICATION_POD_CLEANUP_ENABLED: ConfigEntry[Boolean] =
+    buildConf("kyuubi.kubernetes.terminatedApplicationPodCleanupEnabled")
```
Comment (pan3793) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala:1235]: how about `kyuubi.kubernetes.deleteDriverPodOnTermination.enabled`?

File: kyuubi-server/src/main/scala/org/apache/kyuubi/engine/KubernetesApplicationOperation.scala [hunk 2]
```diff
@@ -98,10 +99,24 @@ class KubernetesApplicationOperation extends ApplicationOperation with Logging {
...
     cleanupTerminatedAppInfoTrigger = CacheBuilder.newBuilder()
       .expireAfterWrite(retainPeriod, TimeUnit.MILLISECONDS)
       .removalListener((notification: RemovalNotification[String, ApplicationState]) => {
-        Option(appInfoStore.remove(notification.getKey)).foreach { removed =>
+        Option(appInfoStore.remove(notification.getKey)).foreach { case (kubernetesInfo, removed) =>
+          if (cleanupPod) {
+            try {
+              val kubernetesClient = getOrCreateKubernetesClient(kubernetesInfo)
+              val deletedPod = kubernetesClient.pods().withName(removed.name).delete().isEmpty
+              info(s"[$kubernetesInfo] Operation of delete pod ${removed.name} with" +
```
Comment (pan3793) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/engine/KubernetesApplicationOperation.scala:111]: let's use info for true and warn for false

**Review by pan3793 — APPROVED** [review]

**Comment by turboFei** [comment]
thanks, merged to master and branch-1.8

File: kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala [hunk 3]
```diff
@@ -1231,6 +1231,14 @@ object KyuubiConf {
       .checkValue(_ > 0, "must be positive number")
       .createWithDefault(Duration.ofMinutes(5).toMillis)
 
+  val KUBERNETES_SPARK_DELETE_DRIVER_POD_ON_TERMINATION_ENABLED: ConfigEntry[Boolean] =
+    buildConf("kyuubi.kubernetes.spark.deleteDriverPodOnTermination.enabled")
+      .doc("If set to true then Kyuubi server will delete the spark driver pod after " +
+        s"the application terminates for ${KUBERNETES_TERMINATED_APPLICATION_RETAIN_PERIOD.key}.")
+      .version("1.8.1")
```
Comment (pan3793) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala:1238]: this feature may not mature for 1.8.1, after another thought:

- my previous suggestion is bad, which causes the configuration name misleading due to the Pod delete moment.
- users may want to only delete the Pod completed with exit code 0 but skip those abnormal exited ones, thus they can get more information to debug later.

Comment (pan3793) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala:1238]: let me think again and send a follow-up PR tomorrow(or weekend)

Comment (zwangsheng) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala:1238]: > users may want to only delete the Pod completed with exit code 0 but skip those abnormal exited ones, thus they can get more information to debug later.

Agree with this, maybe we can change this boolean config to enum policy config, such as NONE, ONLY_COMPLETED, ALL

Comment (turboFei) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala:1238]: I see, we need also check whether the ApplicationInfo.name is null.

Comment (turboFei) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala:1238]: for NOT_FOUND, it might cause NPE.

Comment (turboFei) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala:1238]: for NPE, #5718

**Pull request closed without merging**